### PR TITLE
f22-7pm-1 JUN- NEW JOB(BACKEND): UPDATE SUBJECTS WITH QUARTER ONLY

### PR DESCRIPTION
**Pull request by junhwanlee2316**
ISSUE #22 Backend Part: 
Overview:

New backend job that takes only a quarter as it's parameter, not a quarter and subjectArea

Add new backend job that takes only a quarter as it's parameter, not a quarter and subjectArea
Inside the job logic, get the list of subject areas, and the iterate through those, making one API call at a time, and then updating the MongoDB database for each one.
It may be helpful to put out a log message for each subject area as it is updated with the number of courses added, updated, and a count of errors.
For testing, you can mock a list of subject areas that is MUCH shorter so that the test log output doesn't have to be super long.
Involves a new Job type, and a new entry in the JobsController.

Heroku QA Link: https://f22-7pm-1-courses.herokuapp.com

New Job Controller Entry to POST update with only parameter 'quarter':
Screen Shot 2022-11-30 at 5 05 15 PM

Updated List of Subjects with POST operation with parameter 'quarter':
Screen Shot 2022-11-30 at 5 06 38 PM

### CODE REVIEW

**Review by masonma21 — DISMISSED** [review]
LGTM！

**Review by Abelatnafu — DISMISSED** [review]
LGTM!

File: src/test/java/edu/ucsb/cs156/courses/jobs/UpdateCourseDataJobQuartersTests.java [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+//import org.springframework.beans.factory.annotation.Autowired;
+//import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+//import org.springframework.context.annotation.Import;
```
Comment (pconrad) [inline src/test/java/edu/ucsb/cs156/courses/jobs/UpdateCourseDataJobQuartersTests.java]: Can we remove these commented out imports?

File: src/test/java/edu/ucsb/cs156/courses/jobs/UpdateCourseDataJobQuartersTests.java [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+//import com.fasterxml.jackson.core.type.TypeReference;
+//import com.fasterxml.jackson.databind.ObjectMapper;
```
Comment (pconrad) [inline src/test/java/edu/ucsb/cs156/courses/jobs/UpdateCourseDataJobQuartersTests.java]: And these?

File: src/test/java/edu/ucsb/cs156/courses/jobs/UpdateCourseDataJobQuartersTests.java [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+
+import edu.ucsb.cs156.courses.collections.ConvertedSectionCollection;
+import edu.ucsb.cs156.courses.documents.ConvertedSection;
+import edu.ucsb.cs156.courses.documents.CoursePage;
+import edu.ucsb.cs156.courses.documents.CoursePageFixtures;
+//import edu.ucsb.cs156.courses.documents.Section;
```
Comment (pconrad) [inline src/test/java/edu/ucsb/cs156/courses/jobs/UpdateCourseDataJobQuartersTests.java]: Same

**Review by pconrad — CHANGES_REQUESTED** [review]
A few commented out imports to remove.

Otherwise looks good, and I'll test it on QA

File: src/test/java/edu/ucsb/cs156/courses/jobs/UpdateCourseDataJobFactoryQuartersTests.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
+package edu.ucsb.cs156.courses.jobs;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+
+import org.junit.jupiter.api.Test;
+//import org.junit.jupiter.api.extension.ExtendWith;
+//import org.mockito.Mock;
```
Comment (pconrad) [inline src/test/java/edu/ucsb/cs156/courses/jobs/UpdateCourseDataJobFactoryQuartersTests.java]: commented out imports...

**Review by pconrad — CHANGES_REQUESTED** [review]
Please remove the commented out imports

File: src/test/java/edu/ucsb/cs156/courses/jobs/UpdateCourseDataJobFactoryQuartersTests.java (hunk 4 above)
Comment (junhwanlee2316) [reply src/test/java/edu/ucsb/cs156/courses/jobs/UpdateCourseDataJobFactoryQuartersTests.java]: Fixed commented out imports in the latest commit

**Review by Abelatnafu — APPROVED** [review]
Comments are gone. LGTM!

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**